src/spetlr/autoloader/autoloader_handle.py

```diff
@@ -0,0 +1,66 @@
+from pyspark.sql import DataFrame
+
+from spetlr.configurator.configurator import Configurator
+from spetlr.delta.delta_handle import DeltaHandleInvalidFormat
+from spetlr.exceptions import SparkVersionNotSupportedForSpetlrStreaming
+from spetlr.spark import Spark
+from spetlr.tables import TableHandle
+
+
+class AutoloaderHandle(TableHandle):
+    def __init__(
+        self,
+        *,
+        file_location: str,
+        schema_location: str,
+        data_format: str,
+    ):
+        """
+        file_location: the path to the file location of the delta table
+        schema_location: The location of the cloudfile schema.
+            Databricks documentation uses the expected writeStream checkpoint_path
+            as schema_location, see documentation at
+            https://docs.databricks.com/getting-started/etl-quick-start.html#auto-loader
+
+        data_format: The expected data format extracted by the Autoloader.
+            Examples: JSON, CSV, PARQUET, AVRO, ORC, TEXT, and BINARYFILE.
+            See: https://docs.databricks.com/ingestion/auto-loader/index.html
+        """
+
+        if Spark.version() < Spark.DATABRICKS_RUNTIME_10_4:
+            raise SparkVersionNotSupportedForSpetlrStreaming()
+
+        self._location = file_location
+        self._data_format = data_format
+
+        self._schema_location = schema_location
+
+        self._validate()
+
+    @classmethod
+    def from_tc(cls, id: str) -> "AutoloaderHandle":
+        tc = Configurator()
+        return cls(
+            file_location=tc.table_property(id, "path", ""),
+            data_format=tc.table_property(id, "format", ""),
+            schema_location=tc.table_property(id, "schema_location", ""),
+        )
+
+    def _validate(self):
+        """
+        Validates the dataformat is not delta,
+        since delta stream should use the DeltaHandle instead.
+        """
+        if self._data_format == "delta":
+            raise DeltaHandleInvalidFormat("Use DeltaHandle.read_stream() for delta.")
+
+    def read_stream(self) -> DataFrame:
+        reader = (
+            Spark.get()
+            .readStream.format("cloudFiles")
+            .option("cloudFiles.format", self._data_format)
```

Comment:
Suggestion: make it possible to provide custom options in the orchestrator flow and ingesting them here. Example: when I am reading a CSV file do I want the argument "headers" to be `True` or `False`. Currently I have no way of modifying or providing a such argument/option. Hint: `.options(**self._options_dict)`